client.go

```diff
@@ -158,10 +158,16 @@ func (c *client) authenticate() (*AthenzPrincipal, error) {
 
 // authorize returns true if the supplied principal has access to the resource and action. The initial check is done
 // against the zts endpoint. If that is unreachable, the check is retried against the zms endpoint.
-func (c *client) authorize(ctx context.Context, principal string, check AthenzAccessCheck) (bool, error) {
+func (c *client) authorize(ctx context.Context, principal string, check AthenzAccessCheck, config Config) (bool, error) {
 	var authzResponse struct {
 		Granted bool `json:"granted"`
 	}
+
+	if config.UseCache {
+		cache := config.Cache
+		cache.authorize(ctx, principal, check)
```

Comment:
Let's add prints here saying whether the principal was granted or if there were any errors.